@noun-auction/components/NounishAuction.tsx

```diff
@@ -1,102 +1,154 @@
-import { NounishAuctionProvider } from '@noun-auction/providers'
+import { ClassValue } from 'clsx'
+
+import { useToken } from 'hooks/useToken'
+
+import { useEffect, useState } from 'react'
+import { TypeSafeDao } from 'validators/dao'
+
+import { useNounishAuctionQuery } from '@noun-auction'
 import {
   auctionWrapper,
   auctionWrapperVariants,
   bidHistoryWrapper,
-  debugWrapper,
   wrapperHover,
 } from '@noun-auction/styles/NounishStyles.css'
-import { DaoConfigProps } from '@noun-auction/typings'
-import { Accordion, Box, BoxProps, Grid, Separator, Stack } from '@zoralabs/zord'
+import { useInterval } from '@shared'
+import { Box, BoxProps, Grid, Separator } from '@zoralabs/zord'
 
 import { ActiveAuctionRow } from './ActiveAuction/ActiveAuctionRow'
 import { AuctionHistory } from './AuctionHistory'
-import { AuctionRow } from './AuctionRow'
-import { AuctionDebugger } from './Debuggers'
 
 export interface TokenInfoConfig extends BoxProps {
-  /* ~ Move to provider as config object */
   hideThumbnail?: boolean
   hideTitle?: boolean
   hideCollectionTitle?: boolean
-  thumbnailSize?: 'xxs' | 'xs' | 'sm' | 'md' | 'lg' | '100%' | undefined
+  thumbnailSize?: 'xxs' | 'xs' | 'sm' | 'md' | 'lg' | '100%'
   routePrefix?: string
 }
 
 export interface AuctionViewConfig extends TokenInfoConfig {
-  /* View Config ~ Move to provider as config object */
   showAuctionRow?: boolean
   showBidHistory?: boolean
   useInlineBid?: boolean
-  debug?: boolean
   useErrorMsg?: boolean
   showTopBid?: boolean
   showLabels?: boolean
 }
 
-export interface NounishAuctionProps extends AuctionViewConfig, BoxProps {
-  daoConfig: DaoConfigProps
-  tokenId?: string
-  /* Theming */
+export interface NounishAuctionProps extends AuctionViewConfig {
+  dao: TypeSafeDao
   layout?: keyof typeof auctionWrapperVariants['layout']
 }
 
-export function NounishAuction({
+export function NounishAuction({ dao, ...props }: NounishAuctionProps) {
+  const { activeAuction } = useNounishAuctionQuery({
+    collectionAddress: dao.collectionAddress,
+  })
+
+  const [auctionCompleted, setAuctionCompleted] = useState(false)
+  const ttl = activeAuction ? Date.now() - parseInt(activeAuction.endTime) * 1000 : 0
+
+  useInterval(
+    () => {
+      ttl > 0 ? setAuctionCompleted(true) : setAuctionCompleted(false)
+    },
+    ttl > 0 ? ttl : 0
+  )
+
+  useEffect(() => {
+    if (activeAuction?.winner) setAuctionCompleted(true)
+  }, [activeAuction?.winner])
+
+  if (activeAuction) {
+    const minBidIncrementPercentage = activeAuction?.minBidIncrementPercentage
+    const highestBid = activeAuction?.highestBidPrice?.nativePrice?.raw || '0'
+    const highestBidder = activeAuction?.highestBidder
+    const tokenId = activeAuction?.tokenId
+    const collectionAddress = activeAuction?.collectionAddress
+    const auctionContractAddress = activeAuction?.address
+    const auctionEndTime = activeAuction?.endTime
```

Comment:
This kind of repetitive parsing is a perfect fit for a useNounishAuctionHelper() hook:

```const {highestBid, highestBidder, ... } = useNounishAuctionHelper()```